VirtualList/tests/VirtualGridList-translate-specs.js

```diff
@@ -0,0 +1,801 @@
+		onScrollStopCount,
+		renderItem,
+		resultScrollLeft,
+		resultScrollTop,
+		startScrollTop,
+		svgGenerator;
+
+	beforeEach(() => {
+		clientSize = {clientWidth: 1280, clientHeight: 720};
+		dataSize = 200;
+		items = [];
+		itemSize = {minWidth: 300, minHeight: 240};
+		onScrollCount = 0;
+		onScrollStartCount = 0;
+		onScrollStopCount = 0;
+		resultScrollLeft = 0;
+		resultScrollTop = 0;
+		startScrollTop = 0;
+
+		getScrollTo = (scrollTo) => {
+			myScrollTo = scrollTo;
+		};
+		handlerOnScroll = () => {
+			onScrollCount++;
+		};
+		handlerOnScrollStart = (e) => {
+			startScrollTop = e.scrollTop;
+			onScrollStartCount++;
+		};
+		handlerOnScrollStop = (done, testCase) => (e) => {
+			onScrollStopCount++;
+			resultScrollLeft = e.scrollLeft;
+			resultScrollTop = e.scrollTop;
+
+			testCase();
+			done();
+		};
+
+		renderItem = ({index, ...rest}) => { // eslint-disable-line enact/display-name
+			const {text, subText, source} = items[index];
+
+			return (
+				<ImageItem {...rest} label={subText} src={source}>
+					{text}
+				</ImageItem>
+			);
+		};
+
+		svgGenerator = (width, height, bgColor, textColor, customText) => (
+			`data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 ${width} ${height}' width='${width}' height='${height}'%3E` +
+			`%3Crect width='${width}' height='${height}' fill='%23${bgColor}'%3E%3C/rect%3E` +
+			`%3Ctext x='50%25' y='50%25' dominant-baseline='middle' text-anchor='middle' font-size='36px' fill='%23${textColor}'%3E${customText}%3C/text%3E%3C/svg%3E`
+		);
+
+		const itemNumberDigits = dataSize > 0 ? (dataSize - 1 + '').length : 0;
+		const headingZeros = Array(itemNumberDigits).join('0');
+
+		for (let i = 0; i < dataSize; i++) {
+			const count = (headingZeros + i).slice(-itemNumberDigits),
+				text = `Item ${count}`,
+				subText = `SubItem ${count}`,
+				color = Math.floor(Math.random() * (0x1000000 - 0x101010) + 0x101010).toString(16),
+				source = svgGenerator(300, 300, color, 'ffffff', `Image ${i}`);
+
+			items.push({text, subText, source});
+		}
+	});
+
+	afterEach(() => {
+		clientSize = null;
+		dataSize = null;
+		getScrollTo = null;
+		handlerOnScroll = null;
+		handlerOnScrollStart = null;
+		handlerOnScrollStop = null;
+		items = null;
+		itemSize = null;
+		myScrollTo = null;
+		onScrollCount = null;
+		onScrollStartCount = null;
+		onScrollStopCount = null;
+		renderItem = null;
+		resultScrollLeft = null;
+		resultScrollTop = null;
+		startScrollTop = null;
+		svgGenerator = null;
+	});
+
+	test('should render a list of \'items\'', () => {
+		render(
+			<VirtualGridList
+				clientSize={clientSize}
+				dataSize={dataSize}
+				itemRenderer={renderItem}
+				itemSize={itemSize}
+				scrollMode="translate"
+			/>
+		);
+
+		const expected = 'Item 000SubItem 000';
+		const actual = screen.getByRole('list').children.item(0).textContent;
+
+		expect(actual).toBe(expected);
+	});
+
+	test('should render overhang items when clientSize is not given', () => {
```

Comment:
Is this description correct?

Reply:
I will change it to

'cannot render items when \'clientSize\' and outer DOM size are not specified'